## Introduction

### Background

Microbiome in human gut has increasingly become a research interest, providing us new understandings towards evolution and health. Methods in the field of genomics have been applied to the gut microbiome, such as metagenomics, a subfield of genomics which has came into being for almost two decades {{Gritz 2015}}, and the more recent single-cell approach featuring cell isolation and whole genome amplification from single DNA molecules. {{Shapiro 2013}}

The modern genetics and genomics make use of the new DNA sequencing methods, notably the high-throughput next-generation sequencing. {{Hodkinson 2014}} The tremendous amount of data generated, incompleteness and artifacts due to biochemistry, noises, etc. require computational efforts in problem-solving. The computational aspect of solving biological problems, known as bioinformatics, concerns algorithm design, data science methods, engineering strategies and visualization techniques for the field of microbiome genomics. These methods are necessary to make revealing biological conclusions possible with the data produced by the experiments.

This study discovers genome bins from the massive single-cell and metagomics human gut DNA sequencing data acquired from stool samples by the state-of-the-art experiment design {{TO BE EXPANDED, WITH NUMBERS}}. The thesis will present the data processing and analysis pipeline with application of data science methods, computational strategy design and a few algorithm implementation, as well as the results and the conclusions we reach.

For detailed descriptions about the biological concepts, backgrounds, computational tasks and methods involved, please refer to part II of the thesis and then the additional review chapter.

### Scope & Objectives

#### Subject of the Research



#### Major Tasks



- Discovering genomes from single-cell sequencing data
  - Taxonomic classification of the single-cell sequencing
  - Co-assembly
  - Refining the draft assemblies
- Metagenomics data binning
  - Split and binning by groups
  - Combining binning results
  - Reduce bin redundancy and refining the genome bins
- Assessment
  - Comparing the results of single-cell, metagenomics and combined methods
  - Taxonomic analysis

#### A Brief Profile of the Analyzed Data

##### Mock Community



##### Pilot Single Cell Data



##### Single Cell Data



##### Metagenomics Data



### Structure of the Thesis

